### ISSUE TITLE
mpl not defined for deployed servers

### ISSUE THREAD

**Issue by maxtheman** [marimo-team/marimo#2848]
### Describe the bug

Hello, I'm very much enjoying Marimo so far, and I wanted to report this bug with mo.mpl.

It doesn't seem to exist when I deploy it. 

EXPECTED:
Using: `mo.mpl.interactive(...)` should display a chart in app mode when run through docker.

ACTUAL:
The app polls an incorrect port (10000). When exposed, it doesn't provide any additional CSS or JS, and so 'mpl' is undefined. 
This does not provide any stack trace on the Python side. 
![Image](https://github.com/user-attachments/assets/fb1790fe-db8e-40b3-bede-f5abb89ae9cb)

ADDITIONAL:
I'm using uv, so the output of the environment is from `uv run marimo env`

code: https://github.com/maxtheman/causal_quartets
deployed app: [causal-quartets.fly.dev](https://causal-quartets.fly.dev/)

### Environment

{
  "marimo": "0.9.17",
  "OS": "Darwin",
  "OS Version": "23.6.0",
  "Processor": "arm",
  "Python Version": "3.12.5",
  "Binaries": {
    "Browser": "130.0.6723.117",
    "Node": "v16.17.0"
  },
  "Dependencies": {
    "click": "8.1.7",
    "docutils": "0.21.2",
    "itsdangerous": "2.2.0",
    "jedi": "0.19.1",
    "markdown": "3.7",
    "narwhals": "1.13.3",
    "packaging": "24.2",
    "psutil": "6.1.0",
    "pygments": "2.18.0",
    "pymdown-extensions": "10.12",
    "pyyaml": "6.0.2",
    "ruff": "0.7.3",
    "starlette": "0.41.2",
    "tomlkit": "0.13.2",
    "typing-extensions": "missing",
    "uvicorn": "0.32.0",
    "websockets": "12.0"
  },
  "Optional Dependencies": {
    "pandas": "2.2.3"
  }
}

### Code to reproduce

see github repo

**Comment by maxtheman** [comment]
It seems to also want to poll localhost:10k in Docker on fly.io too, I realize. Maybe there's an environmental variable I missed somewhere?

**Comment by mscolnick** [comment]
I think i know the issue. We spin up a new starlette server in another thread. This works locally since it is still on local host and all ports are exposed, but this does not work when deployed since 1) we are using the internal host instead of the external facing host, and 2) the port is likely not exposed by default. 

I think we try to have this matplotlib backend server hook into the main starlette app to receive these static files

**Comment by maxtheman** [comment]
@mscolnick Just to clarify, it doesn't work for any docker server setup at all even when running locally with an exposed port 10000. I think it's because of the internal host issue you mentioned, from what you're describing. If that was already clear from my initial message, sorry for over clarifying.

If there's some way to specify which host to use for that second thread, I'm happy to do that and expose that port personally. I think the solution you're proposing is probably more elegant and less configuration on the developer's end, but I'm happy to just specify a host if that's an option!

Thank you for the response.

**Comment by mscolnick** [comment]
Hey @maxtheman - yea it wont work in any Docker situation (even non deployed). There is nothing you can set now, but if you'd like to update the code to read from an environment variable, we would gladly accept the contribution. The logic exists here: https://github.com/marimo-team/marimo/blob/main/marimo/_plugins/stateless/mpl/_mpl.py#L204-L205

**Comment by maxtheman** [comment]
@mscolnick I just submitted a PR. It doesn't quite work, Once the host and SSL is sorted out correctly, I run into a subsequent bug that I don't understand the meaning of.

It might be better just to proxy through the main server, like I think your original intention was, but I wanted to submit the WIP so you can see what doesn't work at least.

**Comment by mscolnick** [comment]
Fixed by #2848 (thanks @maxtheman)

Docs: 
````markdown
### Using Interactive Matplotlib Plots in Docker

If your notebook uses `mo.mpl.interactive()` for interactive matplotlib plots, you'll need to make some additional configurations in your Dockerfile:

```Dockerfile
# Add these lines before the CMD instruction
ENV MARIMO_MPL_HOST="0.0.0.0" # same as main host from `marimo run`.
ENV MARIMO_MPL_SECURE=true # if you are running on https
EXPOSE 10000  # Expose in addition to your main application port
```

This exposes the additional CSS and JavaScript required.
````

**Comment by maxtheman** [comment]
Thanks @mscolnick! For posterity, I still haven't been able to get it working on HTTPS, but it does now work on http, and this specific error is resolved either way.

I'll join the discord to discuss.
